pandas/_libs/algos.pyx

```diff
@@ -824,7 +824,7 @@ def rank_1d(
         if values.dtype != np.object_:
             values = values.astype('O')
     else:
-        values = np.asarray(in_arr)
+        values = np.asarray(in_arr).copy()
```

Comment:
prob can remove L838-841 as we are always copying no?

Reply:
good idea, updated